Heightmap config file

ShabSendra opened the pull request:
A Singleton Config class that reads the config file for the heightmap information (path, width, height, scale)

ShabSendra committed:
ajout de la classe Singleton Config, exception a gerer

ShabSendra committed:
ajout de la classe Singleton Config, exception a gerer

ShabSendra committed:
Fixed exception problems, tested with empty file, wrong file, and wrong file path

ShabSendra committed:
Fixed exception problems, tested with empty file, wrong file, and wrong file path

ShabSendra committed:
added smart pointer singleton implementation

ShabSendra committed:
added smart pointer singleton implementation

ShabSendra committed:
changed a include

ShabSendra committed:
changed a include

ShabSendra committed:
remove an include from main

ShabSendra committed:
remove an include from main

ShabSendra committed:
replace tabs whit spaces

ShabSendra committed:
replace tabs whit spaces

PereKastor commented on Projets/FieldConverter/FieldConverter/Config.hpp:
```diff
@@ -0,0 +1,53 @@
+#include <stdexcept>
+#include <iostream>
+#include <memory>
+
```
@SunlayGGX : Petite question qui me vient a l'esprit, les includes doivent se faire dans le .h ou dans le .cpp ? Je ne vois pas forcément leur utilités dans le .h :)

SunlayGGX replied:
Dans la mesure du possible, dans le .cpp. 
On met dans le .h les includes que lorsque l'on a déclaré une méthode prototype utilisant une valeur, ou quand on est un codeur atteint de flemme du "je n'ai pas envie de déclarer une classe que j'implémenterai plus tard", ou quand on a pas le choix. Cependant, les inclure dans le .h peut potentiellement avoir 2 impacts majeurs : Le temps de compilation et la taille du code qui augmente exponentiellement...

Ici, ShabSendra n'a pas eu le choix que d'inclure string, memory et stdexcept (car c'est très difficile de déclarer existant string et stdexcept quand l'un est un typedef d'un template et que de plus, il les utilise explicitement par valeur (pour unique_pointer et string)).

Cependant, fstream, sstream et iostream peuvent être plaçés dans le .cpp car on utilise pas leur type ici.

ShabSendra replied:
En effet, erreur de ma part c'est corrigé

PereKastor committed:
Fix all conflicts between branch before merge

Merge branch 'heightmap-config-file' of github.com:PirateTeamDDJV12/TeamWorkPirate into heightmap-config-file

# Conflicts:
#	Projets/FieldConverter/FieldConverter/FieldConverter.vcxproj.filters
#	Projets/FieldConverter/FieldConverter/main.cpp

ShabSendra committed:
Fixed useless includes in .hpp added a try catch in main to prevent the program to crash in case of wrong config file

ShabSendra committed:
fix .filters visual studio file

ShabSendra committed:
fix tab to spaces
